### PR TITLE
fix(Modal): custom footer render error

### LINKED ISSUE #44923: Modal  "自定义页脚渲染函数"示例显示有误
### Reproduction link

[https://ant-design.antgroup.com/components/modal-cn#modal-demo-footer-render](https://ant-design.antgroup.com/components/modal-cn#modal-demo-footer-render)

### Steps to reproduce

1. 打开组件文档中 Modal下 “自定义页脚渲染函数”示例，"取消"和"确认"按钮显示有误

### What is expected?

按钮正常显示

### What is actually happening?

这两个按钮显示空白

| Environment | Info |
| --- | --- |
| antd | 5.9.1 |
| React | * |
| System | Window |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fixed: #44923

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fix(Modal): custom footer render errors      |
| 🇨🇳 Chinese |     修复模态框自定义 footer 时显示错误      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 34d8b83</samp>

Added support for custom footer functions in the `Modal` component, which can receive the modal state and the default buttons as arguments. Added tests and a demo example for this feature. Refactored some code to avoid TypeScript errors.

### <samp>🤖 Generated by Copilot at 34d8b83</samp>

* Add a custom footer function prop to the `Modal` component that allows users to customize the footer content and access the default Ok and Cancel buttons as React components ([link](https://github.com/ant-design/ant-design/pull/44929/files?diff=unified&w=0#diff-28fe880db90aca35d37607b3087393a1815128428f281a9a8394d5dc34475371L74-R79),[link](https://github.com/ant-design/ant-design/pull/44929/files?diff=unified&w=0#diff-28fe880db90aca35d37607b3087393a1815128428f281a9a8394d5dc34475371R86-R87),[link](https://github.com/ant-design/ant-design/pull/44929/files?diff=unified&w=0#diff-9103d37c6668043c170813d54cb43ca28bc6cf111eedfc6718aeae7f540a0b27L63-R63))
* Test the custom footer function prop in `Modal.test.tsx` by checking its parameters, return value, and DOM output ([link](https://github.com/ant-design/ant-design/pull/44929/files?diff=unified&w=0#diff-5b32c615603a02b35bc7b0b1651031eb23c5af3d939495dc127088d2efcd58b0R138-R149),[link](https://github.com/ant-design/ant-design/pull/44929/files?diff=unified&w=0#diff-5b32c615603a02b35bc7b0b1651031eb23c5af3d939495dc127088d2efcd58b0R165-R201))
* Demonstrate the custom footer function prop in `render-panel.tsx` by showing an example with a custom button and a different order of the default buttons ([link](https://github.com/ant-design/ant-design/pull/44929/files?diff=unified&w=0#diff-93d352b81d4be73b9052974d0ac2171fd5d5c9ec35ad5765511cc21078e3b081L2-R2),[link](https://github.com/ant-design/ant-design/pull/44929/files?diff=unified&w=0#diff-93d352b81d4be73b9052974d0ac2171fd5d5c9ec35ad5765511cc21078e3b081R18-R35))

### CODE REVIEW

File: components/modal/PurePanel.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ const PurePanel: React.FC<PurePanelProps> = (props) => {
     additionalProps = {
       closable: closable ?? true,
       title,
-      footer: props.footer === undefined ? <Footer {...props} /> : props.footer,
+      footer: props.footer !== null ? <Footer {...(props as any)} /> : props.footer,
```
Comment (zombieJ) [inline components/modal/PurePanel.tsx:63]: 这个没看懂，`props.footer === null` 的时候传入 `props.footer` 是为什么...

Comment (kiner-tang) [reply components/modal/PurePanel.tsx]: +1

Comment (Wxh16144) [reply components/modal/PurePanel.tsx]: CV 的这里逻辑 https://github.com/ant-design/ant-design/blob/b24a6465f12d98f191e3952a57d70017c34b53bd/components/modal/Modal.tsx#L93-L95

Comment (Wxh16144) [reply components/modal/PurePanel.tsx]: 改这个 `<PurePanel />` 是为了给 argos 做 截图测试

**Comment by afc163** [comment]
lint 挂了

**Review by afc163 — APPROVED** [review]

File: components/modal/__tests__/Modal.test.tsx [hunk 2]
```diff
@@ -150,4 +162,41 @@ describe('Modal', () => {
...
+            <OkBtn />
+            <CancelBtn />
+          </>
+        )}
+      />,
+    );
+
+    expect(firstArg()).toEqual(secondArg());
+  });
+
```
Comment (Wxh16144) [inline components/modal/__tests__/Modal.test.tsx:182]: 这条 case 不生效，删除了再 merge 吧。 @afc163 
```suggestion
```

**Pull request merged**